librz/core/canalysis.c

```diff
@@ -6726,4 +6726,130 @@ RZ_API bool rz_core_analysis_esil_trace_stop(RzCore *core) {
 	esil->trace = NULL;
 	rz_config_set_i(core->config, "dbg.trace", false);
 	return true;
-}
\ No newline at end of file
+}
+
+/**
+ * Free RzAnalysisBytes
+ *
+ * @param ptr RzAnalysisBytes pointer
+ */
+RZ_API void rz_analysis_bytes_free(void *ptr) {
+	rz_return_if_fail(ptr);
+	RzAnalysisBytes *ab = ptr;
+	rz_analysis_op_free(ab->op);
+	ab->op = NULL;
+	rz_analysis_hint_free(ab->hint);
+	ab->hint = NULL;
+	RZ_FREE(ab->opcode);
+	RZ_FREE(ab->disasm);
+	RZ_FREE(ab->pseudo);
+	RZ_FREE(ab->description);
+	RZ_FREE(ab->mask);
+	RZ_FREE(ab->bytes);
+	RZ_FREE(ab);
+}
+
+/**
+ *
+ * Analysis and disassemble bytes use rz_analysis_op and rz_asm_disassemble
+ *
+ * @param core The RzCore instance
+ * @param buf data to analysis
+ * @param len analysis len bytes
+ * @param nops analysis n ops
+ * @return list of RzAnalysisBytes
+ */
+RZ_API RZ_OWN RzList *rz_core_analysis_bytes(RzCore *core, const ut8 *buf, int len, int nops) {
+	bool be = core->print->big_endian;
+	core->parser->subrel = rz_config_get_i(core->config, "asm.sub.rel");
+	int ret, i, idx;
+	RzAsmOp asmop;
+	ut64 addr;
+	RzList *list = rz_list_newf(rz_analysis_bytes_free);
+	RzAnalysisBytes *ab;
+
+	for (i = idx = 0; idx < len && (!nops || (nops && i < nops)); i++, idx += ret) {
+		ab = RZ_NEW0(RzAnalysisBytes);
+		rz_list_append(list, ab);
+		addr = core->offset + idx;
+		rz_asm_set_pc(core->rasm, addr);
+		ab->hint = rz_analysis_hint_get(core->analysis, addr);
+		ab->op = rz_analysis_op_new();
+		ret = rz_analysis_op(core->analysis, ab->op, addr, buf + idx, len - idx,
+			RZ_ANALYSIS_OP_MASK_ESIL | RZ_ANALYSIS_OP_MASK_IL | RZ_ANALYSIS_OP_MASK_OPEX | RZ_ANALYSIS_OP_MASK_HINT);
+
+		(void)rz_asm_disassemble(core->rasm, &asmop, buf + idx, len - idx);
+		ab->opcode = rz_asm_op_get_asm(&asmop);
+
+		char *mnem = strdup(rz_asm_op_get_asm(&asmop));
+		char *sp = strchr(mnem, ' ');
+		if (sp) {
+			*sp = 0;
+			if (ab->op->prefix) {
+				char *arg = strdup(sp + 1);
+				sp = strchr(arg, ' ');
+				if (sp) {
+					*sp = 0;
+				}
+				free(mnem);
+				mnem = arg;
+			}
+		}
+		if (ret < 1) {
+			RZ_LOG_ERROR("Invalid instruction at 0x%08" PFMT64x "...\n", addr);
+			free(mnem);
+			rz_list_free(list);
+			return NULL;
+		}
```

Comment:
I think invalid instructions should appear as "pseudo-instructions" and this check should be removed.

Reply:
pseud-instructions as `invalid`

Reply:
Are there any examples?

Reply:
if rz_asm_disassemble < 1 {
   ab->opcode = "invalid"
   mnem = "invalid"
   ab->bytes = rz_hex_bin2strdup(buf + idx, min_size);
   ....
}